### DISCUSSION TITLE
Pydantic version validation

### DISCUSSION THREAD

**Discussion by david-shiko** [pydantic/pydantic#4412] [Question] [answered]
Is pydantic has built-in validation for version string? 
Like "major.minor.micro+some_additional_text"

I'm asking because many projects uses this convention (for example "poetry")
I'm don't know exactly RFC for this convention but [pep 440](https://peps.python.org/pep-0440/#final-releases) partially describing about this.

**Comment by samuelcolvin** [comment, marked as answer]
Are you asking if there's a validator for python versions?

We have no builtin validator, but could could build one using `pkg_resources.safe_version`

**Reply by david-shiko** [reply]
Thanks for a quick reply!
No, I mean all versions, not only python.
IIRC the pattern for version  "major.minor.micro+some_additional_text" is common for all languages and versions.

**Reply by samuelcolvin** [reply]
No, it's slightly different for different languages, e.g. rust uses `1.0-alpha1` while python would use `1.0.0a1`, differences:
* whether to include the patch version when it's `0`
* whether to use name or initial of `alpha`, `beta` `post` releases etc.

There are probably more differenecs.

See [this comment](https://github.com/pydantic/pydantic-core/blob/59babff00dd2b19130800dc6e55bbaccc7b9d752/src/lib.rs#L24-L28)

**Reply by david-shiko** [reply]
By version convention I mean this article: https://semver.org/. This is a global convention.
Probably python and rust version samples above are "language specific local convention".

P.S. As I see, mentioning link to the pep 440 was a wrong example :)
